inc/blocks/namespace.php

```diff
@@ -157,6 +160,90 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 	}
 }
 
+/**
+ * Synchronise any XBs on the widget page with a shadow post type.
+ *
+ * @param array $instance The current widget settings.
+ * @param array $new_instance The of new widget settings.
+ * @param array $old_instance The of old widget settings.
+ * @param object $widget The current widget instance.
+ * @return array $instance The current widget settings.
+ */
+function on_widgets_save( $instance, $new_instance, $old_instance, $widget ) {
+
+	if ( ! isset( $instance['content'] ) ) {
+		return $instance;
+	}
+
+	// Scan for XBs in the post content.
+	$blocks = parse_blocks( $instance['content'] );
+	$xbs = find_xbs( $blocks );
+
+	// Find referenced XBs.
+	$existing_posts = new WP_Query( [
+		'post_type' => POST_TYPE,
+		'post_status' => 'any',
+		'post_name__in' => array_filter( array_map( function ( array $xb ) : string {
+			return $xb['attrs']['clientId'];
+		}, $xbs ) ),
+		'posts_per_page' => 100,
+		'no_found_rows' => true,
+	] );
+
+	// Update or create XB posts.
+	foreach ( $xbs as $index => $xb ) {
+		if ( empty( $xb['attrs']['clientId'] ) ) {
+			continue;
+		}
+
+		// Extract existing post if found.
+		$posts = wp_list_filter( $existing_posts->posts, [
+			'post_name' => $xb['attrs']['clientId'],
+		] );
+		$posts = array_values( $posts ); // Reset keys.
+
+		// Generate a default using the current post title and instance number in the content.
+		$default_title = sprintf( __( 'Widget %s %d', 'altis-analytics' ), ucfirst( Blocks\get_block_type( $xb['attrs']['clientId'] ) ), $widget->number );
```

Comment:
The `get_block_type()` function is in the current namespace so you don't need to use the `Blocks` namespace prefix.
```suggestion
		$default_title = sprintf( __( 'Widget %s %d', 'altis-analytics' ), ucfirst( get_block_type( $xb['attrs']['clientId'] ) ), $widget->number );
```

What are the outputs from this function and what are the results when using `ucfirst`? Would it look ok to end users?

For example see the following: https://github.com/humanmade/aws-analytics/blob/master/inc/dashboard/namespace.php#L216-L230

If you rebase against `master` you could add a `get_block_type_name()` function perhaps or refactor `get_block_type()` to add an extra parameter to called "output" like `get_block_type( $client_id, 'name' )` for example.

Reply:
The `ucffirst` was making the title render as "Widget Personalization 22" which I suspect I can remove a and leave it as "Widget personalization 22" ?